### PR TITLE
Enable Istio build on Arm64 platform

### PR DESCRIPTION
**Pull request by Jingzhao123**
1. The Makefile.core.mk is modified for adapting multi-arch
   platform.
2. In init.sh scripts, replace the "amd64" with "GOARCH"
3. Add arch parameter in dockerfile for choosing build
   architecture.
4. Add build options for building multi-arch images in
   istio-docker.mk file
5. Add arm64 option for helm template in values.yaml

Signed-off-by: Jingzhao <Jingzhao.Ni@arm.com>

### CODE REVIEW

File: install/kubernetes/helm/istio/values.yaml [hunk 1]
```diff
@@ -383,6 +383,7 @@ global:
     amd64: 2
     s390x: 2
     ppc64le: 2
+    arm64: 2
```
Comment (howardjohn) [inline install/kubernetes/helm/istio/values.yaml]: Won't we have  an amd64 image and an arm64 image? So now our amd64 images may be scheduled on arm64 nodes?

File: pilot/docker/Dockerfile.proxyv2 [hunk 2]
```diff
@@ -19,10 +19,13 @@ COPY istio-iptables /usr/local/bin/istio-iptables
...
 FROM gcr.io/distroless/cc:latest as distroless
 
+# The default value of arch is x86_64
+ARG ARCH=x86_64
+
 # TODO(https://github.com/istio/istio/issues/17656) clean up this hack
 COPY --from=default /sbin/xtables-multi /sbin/iptables* /sbin/ip6tables* /sbin/ip /sbin/
-COPY --from=default /usr/lib/x86_64-linux-gnu/xtables/ /usr/lib/x86_64-linux-gnu/xtables
-COPY --from=default /usr/lib/x86_64-linux-gnu/ /usr/lib/x86_64-linux-gnu
+COPY --from=default /usr/lib/${ARCH}-linux-gnu/xtables/ /usr/lib/${ARCH}-linux-gnu/xtables
```
Comment (howardjohn) [inline pilot/docker/Dockerfile.proxyv2]: Does this really work?

Comment (Jingzhao123) [reply pilot/docker/Dockerfile.proxyv2]: I have successfully built the corresponding images on x86 & arm64 with following command. So, I think it is OK.
```
make build
make docker
```

**Comment by howardjohn** [comment]
You can build them but do they run properly?

File: install/kubernetes/helm/istio/values.yaml (hunk 1 above)
Comment (Jingzhao123) [reply install/kubernetes/helm/istio/values.yaml]: yes, there is no arm64 image. So i do some modification for supporting build arm64 images.

**Comment by Jingzhao123** [comment]
Hi @howardjohn 
The purpose of this patch only focuses on building Istio on Arm64. If want to run the istio on Arm64, it still need to do more works, just like supporting multi-arch docker image, updating envoy component. So, in this patch, i only do some modifications on Makefile and Dockerfile. I will upstream other patches for further porting work.

**Comment by Jingzhao123** [comment]
Hi @howardjohn 
         I have been build the 1.13 K8s environments for testing Istio on x86. Could you please tell me how to verify the Istio images and configurations ? If those images could be run correctly on it, does it means that those images are OK? 
         On the other hand, for arm64 platform, i just want to enable to build istio, but not running it (actually, there may be lots of work to do). So, I will not verify it on it. I will do more contributions about it. Is this feasible?

**Comment by howardjohn** [comment]
> On the other hand, for arm64 platform, i just want to enable to build istio, but not running it (actually, there may be lots of work to do). So, I will not verify it on it. I will do more contributions about it. Is this feasible?

I am perfectly happy adding code to get arm64 working, but I am not sure what is the point of building the images if they do not run? Given that adding support for another platform adds some complexity, it would be good to know that it actually works before merging.

**Comment by Jingzhao123** [comment]
Hi, @howardjohn 
            Thanks for your comments. I also agree with your opinion. And i will verify those modification on both platforms. On the other hand, i want to ask a few questions about it.
1. How to verify those functions ? 
    If those images could be run correctly on k8s, does it means that those images are OK?
2. Now, i want to build the arm64 docker images by qemu tools. Is it OK?
    Now, as far as i known, since there is no arm64 platform, Istio arm64 images can not be built on Google cloud. Could i build those image by [qemu tools](https://github.com/multiarch/qemu-user-static) ? Just like Kubernetes.

**Comment by Jingzhao123** [comment]
Hi, @howardjohn 
            Thanks for your comments. I also agree with your opinion. And i will verify those modification on both platforms. On the other hand, i want to ask a few questions about it.
1. How to verify those functions ? 
    If those images could be run correctly on k8s, does it means that those images are OK?
2. Now, i want to build the arm64 docker images by qemu tools. Is it OK?
    Now, as far as i known, since there is no arm64 platform, Istio arm64 images can not be built on Google cloud. Could i build those image by [qemu tools](https://github.com/multiarch/qemu-user-static) ? Just like Kubernetes.

**Pull request closed without merging**